source/core.py

```diff
@@ -560,6 +560,15 @@ def run(self):
 	else:
 		log.debug("initializing updateCheck")
 		updateCheck.initialize()
+	# If running from source, try to disconnect from the console we may have been executed in.
+	# NVDA may reconnect to read it later,
+	# but it is better to assume we are not connected to anything at the start.
+	try:
+		ctypes.windll.kernel32.FreeConsole()

```

Comment:
This code assumes that `FreeConsole` would raise a `WindowsError` if  NVDA wasn't connected to the console but it seems not to be the case. According to the documentation this function returns true if the console has been fried and False if `FreeConsole` fails or there was no connected console in the first place.